### PR TITLE
Add cmd arguments

### PR DESCRIPTION
**Pull request by p00f**
This allows 0 or 1 arguments to `:Neogit`, so `:Neogit` retains default behaviour but `:Neogit floating` now uses a floating window

### CODE REVIEW

**Comment by TimUntersberger** [comment]
Could you change the syntax from `Neogit floating` to `Neogit kind=floating`? Simple parsing is enough, but we need the syntax change to make future subcommands easier to distinguish.

**Comment by p00f** [comment]
Sure!

**Comment by p00f** [comment]
How would it work though, I think you'd have to make `neogit.status.create()` take a table

**Comment by TimUntersberger** [comment]
For now just pass in the value of kind directly

**Comment by p00f** [comment]
Can I do the parsing in the lua function? Vimscript regex is very hard

**Comment by p00f** [comment]
https://user-images.githubusercontent.com/36493671/112026931-b54e6d00-8b5c-11eb-84ca-b914deaebcc7.mp4

**Comment by p00f** [comment]
(I did the parsing in the lua function, I couldn't figure out how to match things in vimscript)

**Comment by TimUntersberger** [comment]
> Can I do the parsing in the lua function? Vim regex is very hard

Parsing in lua is 100% prefered. 

Obviously the name I chose for the function is just a placeholder and you can change it to whatever you think fits best.

File: lua/neogit/status.lua [hunk 1]
```diff
@@ -688,8 +688,23 @@ local discard = a.sync(function()
...
+local function create(...)
+  local args = {...}
+  local kind = nil
+
+  for _, val in pairs(args) do
+    local kindpos = string.find(val, "kind=")
+    if kindpos then
+      kind = string.sub(val, 6)
+    end
+  end
```
Comment (TimUntersberger) [inline lua/neogit/status.lua:700]: Sorry, but could you extract the parsing out of the create function and change the signature back to how it is right now? You can then change mapping to:

```lua
require'neogit.status'.create(require'neogit.util'.parse_command_args(<f-args>))
```
Obviously the name I chose for the function is just a placeholder and you can change it to whatever you think fits best.

**Review by TimUntersberger — CHANGES_REQUESTED** [review]

**Review by TimUntersberger — APPROVED** [review]
Thank you!

**Pull request merged**